pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -668,6 +1015,55 @@ func (pl *DynamicResources) PostFilter(ctx context.Context, cs fwk.CycleState, p
 			return nil, fwk.NewStatus(fwk.Unschedulable, "deallocation of ResourceClaim completed")
 		}
 	}
+
+	if extendedResourceClaim != nil && extendedResourceClaim.Name != specialClaimInMemName {
+		// If the special resource claim for extended resource backed by DRA
+		// is reserved or allocated at prior scheduling cycle, then it should be deleted.
+		extendedResourceClaim := extendedResourceClaim.DeepCopy()
+
+		// Remove the finalizer to unblock removal first.
+		builtinControllerFinalizer := slices.Index(extendedResourceClaim.Finalizers, resourceapi.Finalizer)
+		if builtinControllerFinalizer >= 0 {
+			extendedResourceClaim.Finalizers = slices.Delete(extendedResourceClaim.Finalizers, builtinControllerFinalizer, builtinControllerFinalizer+1)
+			retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				_, err := pl.clientset.ResourceV1beta1().ResourceClaims(extendedResourceClaim.Namespace).Update(ctx, extendedResourceClaim, metav1.UpdateOptions{})
+				if err != nil {
+					return fmt.Errorf("update resourceclaim %s/%s: %w", extendedResourceClaim.Namespace, extendedResourceClaim.Name, err)
+				}
+				return nil
+			})
+			if retryErr != nil {
+				return nil, statusError(logger, retryErr)
+			}
+		}
+
+		// TODO: if the claim is allocated or reserved, that must be cleared before it can be deleted. Need integration-level tests for this,
```

Comment:
Sorry, my bad: we don't actually have that protection. I was thinking about adding it at some point, but then didn't. So removing the finalizer + delete is enough to make the ResourceClaim go away after all, even if it has an Allocation and a consumer in ReservedFor while it gets deleted - I just verified that.

You can remove this code again. An actual integration test would still be good, but also is harder to set up.

Reply:
ok, good to know, removed.